core/src/main/scala/cats/data/NonEmptyChain.scala

```diff
@@ -586,10 +586,17 @@ class NonEmptyChainOps[A](private val value: NonEmptyChain[A])
 
 sealed abstract private[data] class NonEmptyChainInstances extends NonEmptyChainInstances1 {
 
-  implicit val catsDataInstancesForNonEmptyChain: SemigroupK[NonEmptyChain]
+  // Required for binary compatibility with v2.6.1.
+  def catsDataInstancesForNonEmptyChain: SemigroupK[NonEmptyChain]
```

Comment:
I was pretty sure that I tried to make it package private in the original PR and it didn't work – it broke binary compatibility for some Scala version (or at least Mima though it does). But I've just given it another try and don't see any errors from Mina neither for Scala 2.12 nor for Scala 2.13. I got confused a little bit :)

On the second thought, I would assume that deprecating it might be slightly better because making it package private potentially (although unlikely) breaks *source* compatibility – in case if someone references this field explicitly in their code. WDYT?

Reply:
Oh, @djspiewak just mentioned in another issue:
> We generally don't consider implicits to be part of source compatibility, so IMO this is something we can (and should) do in 2.7.1

Good to know, thanks. I guess I can safely make it package private then (since it seemingly doesn't break BC).

Reply:
Turns out, `private[data]` works for `catsDataInstancesForNonEmptyChain` only. For three other similar obsolete former implicits it breaks the binary compatibility. So I annotated them as deprecated without changes in their visibility.

Reply:
Were the mima problems only for static methods? I don't know what the policy about static methods is for cats, since those are only used by Java.